utils/hls.py

```diff
@@ -150,34 +151,31 @@ def from_point_pandas(cls, df, bands=[], tile_lookup=None):
         df['sensor'] = df.apply(lambda row: 'L' if 'L30' in row.scene else 'S', axis=1)
         # get datetime for scene
         df['dt'] = df.apply(lambda row: _scene_to_datetime(row.scene), axis=1)
-
         # create xr_dataset
         xr_ds = xr.Dataset.from_dataframe(df)
         xr_ds.attrs['bands'] = bands
         return cls(xr_ds)
 
+
     @classmethod
-    def from_bbox(cls, bbox, years, bands=[], tile_lookup=None):
+    def from_point_pandas(cls, df, bands=[], tile_lookup=None):
+        """
+        Args:
+            df (pandas.DataFrame): Dataframe with at least the following columns: lat, lon, year
+            landsat_bands list(str): list of landsat band names to fetch
+            sentinel_bands list(str): list of landsat band names to fetch
+        """
         lookup = tile_lookup if tile_lookup else HLSTileLookup()
+        # eh?
+        df = df
+        tiles = df.apply(lambda row: lookup.get_point_hls_tile_id(row.lat, row.lon), axis=1)
+        return self.from_tiles(tiles, years, bands, tile_lookup)
```

Comment:
One of the (unstated) goals of this function was to retain metadata provided in the `df` parameter (such as training id). This isn't preserved here because `from_tiles` operates only on a series of tiles.

Also there are no years passed into `from_point_pandas` because it's required that the year is a column in the input df. 

One way we can use your approach is:
1. get the series of tiles for the points and add it as a column to the input `df`.
2. sub-select two columns from `df`: year and tile 
3. remove duplicates the sub-df
4. adapt `from_tiles` to accept a dataframe with tile+year rather than accepting tiles and years separately
5. Call `from_tiles` on the sub-df in (3)
6. Join the resulting dataset from `from_tiles` with the point dataframe on year/tile

This saves us duplicate computation for points with the same tile/year and means we can share code with `from_bbox`

Reply:
Thanks, I see now this didn't work. I really just wanted a `from_tiles` fxn to test with and wanted to reduce duplication, but....

Are you in a situation where you need different years of data for different points?

Reply:
Yeah one dataset has samples from different years